### PR TITLE
Add new methods to socket stubs.

### PR DESCRIPTION
**Pull request by hashstat**
Also fix a couple of invalid types in recv*_into() socket methods.

### CODE REVIEW

File: stdlib/2and3/socket.pyi [hunk 1]
```diff
@@ -559,6 +560,28 @@ class socket:
     def setsockopt(self, level: int, optname: int, value: Union[int, bytes]) -> None: ...
     def shutdown(self, how: int) -> None: ...
 
+    if sys.version_info >= (3, 3):
+        @overload
+        def recvmsg(self, bufsize: int) -> Tuple[bytes, List[_CMSG], int, Any]: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/socket.pyi]: Why do these need overloads? Can't you just use default values?

Comment (hashstat) [reply stdlib/2and3/socket.pyi]: Using default values would mean that type checking wouldn't complain if one did this:

```python
sock.recvmsg(100, flags=socket.MSG_WAITALL)
```

But at run-time the following exception is raised:

```
TypeError: recvmsg() takes no keyword arguments
```

The intention was that overloading creates an error during static type checking so one doesn't have to wait for it to fail at run-time. But now that I'm testing it, the error produced by mypy is unhelpful and one can still do this:

```python
sock.recvmsg(100, 100, flags=5)
```

In which case mypy doesn't complain and the run-time error still occurs. :-(

According to the MyPy cheat sheet, I could append double underscores (__) to the parameters tomake them position-only arguments, but the mypy error is still unhelpful:

```
error: No overload variant of "recvmsg" of "socket" matches argument types [builtins.int, builtins.int, builtins.int]
```

Is there a better way? Or am I overthinking this one.

Comment (JelleZijlstra) [reply stdlib/2and3/socket.pyi]: What if you don't use overloads, but prefix the arguments with double underscores, like `def recvmsg(size: int, __flags: int = ...) -> SomeType: ...`?

In general, we've mostly not worried about positional-only arguments to builtins.

Comment (hashstat) [reply stdlib/2and3/socket.pyi]: Brilliant! Works like a charm. And the error message is meaningful too.

File: stdlib/2and3/socket.pyi [hunk 2]
```diff
@@ -559,6 +560,14 @@ class socket:
     def setsockopt(self, level: int, optname: int, value: Union[int, bytes]) -> None: ...
     def shutdown(self, how: int) -> None: ...
 
+    if sys.version_info >= (3, 3):
+        def recvmsg(self, __bufsize: int, __ancbufsize: Optional[int] = ...,
```
Comment (JelleZijlstra) [inline stdlib/2and3/socket.pyi]: I don't think these are Optional (`sock.recvmsg(1, None)` fails with a TypeError).

Comment (hashstat) [reply stdlib/2and3/socket.pyi]: You are totally right. For some reason I got it in my head that all default arguments needed to be Optional as well. Something about those ellipsis and the statement in the contributing guide regarding being implicit about Optionals. Pretty sure I did the same nonsense in the os module stubs as well. Thanks for catching this.

**Pull request merged**